src/helpers/withLibcore.js

```diff
@@ -0,0 +1,31 @@
+// @flow
+
+const core = require('@ledgerhq/ledger-core')
+
+let walletPool = null
+let queue = Promise.resolve()
+
+// TODO: `core` and `NJSWalletPool` should be typed
+type Job = (Object, Object) => any
+
+export default function withLibcore(job: Job) {
+  if (!walletPool) {
+    walletPool = core.instanciateWalletPool({
+      // sqlite files will be located in the app local data folder
+      dbPath: process.env.LEDGER_LIVE_SQLITE_PATH,
+    })
+  }
+  // $FlowFixMe WTF is happening here, dudes.
```

Comment:
🤣🤣🤣
BTW it would be great if you could define a parametric type so that we have this:

```js
type Job<A> = (*, *) => Promise<A>
export default function<A> withLibcore(job: Job<A>): Promise<A> {
```

that might help type & also it's probably better for type..

Reply:
honestly I tried to do that. and failed to make it works properly. maybe we'll see that together later.

Reply:
sent you a PR